doc/source/whatsnew/v0.25.0.rst

```diff
@@ -741,6 +741,52 @@ consistent with NumPy and the rest of pandas (:issue:`21801`).
    cat.argsort()
    cat[cat.argsort()]
 
+.. _whatsnew_0250.api_breaking.list_of_dict:
+
+Column order is preserved when passing a list of dicts to DataFrame
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Starting with Python 3.7 the key-order of ``dict`` is `guaranteed <https://mail.python.org/pipermail/python-dev/2017-December/151283.html>`_. In practice, this has been true since
+Python 3.6. The :class:`DataFrame`  constructor now treats a list of dicts in the same way as
+it has a list of ``OrderedDict`` since v0.19.0. This change applies only when pandas is running
+on Python>=3.6. As a consequence, the column order produced by :class:`DataFrame`
+in such cases has changed (:issue:`27309`).
+
+.. ipython:: python
+
+   data = [
+       {'name': 'Joe', 'state': 'NY', 'age': 18},
+       {'name': 'Jane', 'state': 'KY', 'age': 19, 'hobby': 'Minecraft'},
+       {'name': 'Jean', 'state': 'OK', 'age': 20, 'finances': 'good'}
+   ]
+
+*Previous Behavior*:
+
+The columns were lexicographically sorted previously,
+
+.. code-block:: python
+
+   In [1]: pd.DataFrame(data)
+   Out[1]:
+      age finances      hobby  name state
+   0   18      NaN        NaN   Joe    NY
+   1   19      NaN  Minecraft  Jane    KY
+   2   20     good        NaN  Jean    OK
+
+*New Behavior*:
+
+The column order now matches the insertion-order of the keys in the ``dict``,
+considering all the records from top to bottom.
+
+.. code-block:: python
```

Comment:
@jreback meant to say that this should also be a `.. ipython:: python` block, and then in that case the prompts are indeed not needed

Reply:
And I said that none of the other segments do that, so why this case? 

Maybe I'm misunderstanding. IIUC `code-block` means "please syntax-highlight", while ipython means "please run and include output". 

If that's correct, why would you want this to ever change based on code changes? and if it's done in previous whatsnew (like the one that introduces sorted keys for `Dataframe(dict)` for example, this PR now breaks *that* documentation. And if this is a code-block, why should the previous snippet be executed by ipython and perhaps also break in the future (like kwd changes)?

In my view, whatsnew should be immutable (for code changes) because it's a specification. And
you seemed to think so at one point too #6856.

Reply:
> And I said that none of the other segments do that, so why this case?

If you look at the rest of the file, you will see that *all* other code blocks for "new behaviour" use `.. ipython:: python` blocks (and the ones showing old behaviour use `code-block`). 
That's how it is currently done. You can certainly disagree with that habit, but then a discussion about that can happen in a new issue, not here. 

> And you seemed to think so at one point too #6856.

That issue is for *old* whatsnew files.

Reply:
ok.

Reply:
> That issue is for *old* whatsnew files.

I looked and the whatsnew for 0.22, 0.23 and 0.24 at least still contain ipython blocks for "new" behavior, so the problem I described is still on point afaict.

Reply:
Therefore, the issue you referenced to convert those, is an *open* issue :-)